maven-plugin-lib/src/main/java/com/microsoft/azure/maven/FTPUploader.java

```diff
@@ -0,0 +1,170 @@
+        }
+
+        if (isSuccess) {
+            log.info(UPLOAD_SUCCESS + ftpServer);
+        } else {
+            throw new MojoExecutionException(UPLOAD_FAILURE);
+        }
+    }
+
+    /**
+     * Upload directory to specified FTP server without retries.
+     *
+     * @param ftpServer
+     * @param username
+     * @param password
+     * @param sourceDirectoryPath
+     * @param targetDirectoryPath
+     * @return Boolean to indicate whether uploading is successful.
+     */
+    protected boolean uploadDirectory(final String ftpServer, final String username, final String password,
+                                      final String sourceDirectoryPath, final String targetDirectoryPath) {
+        log.debug("FTP username: " + username);
+        boolean isSuccess = false;
+        try {
+            ftpClient.connect(ftpServer);
+            ftpClient.login(username, password);
+            ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+            ftpClient.changeWorkingDirectory(targetDirectoryPath);
+
+            log.info(String.format(UPLOAD_DIR_START, sourceDirectoryPath, targetDirectoryPath));
+            uploadDirectory(sourceDirectoryPath, targetDirectoryPath, "");
+            log.info(String.format(UPLOAD_DIR_FINISH, sourceDirectoryPath, targetDirectoryPath));
+
+            isSuccess = true;
+            ftpClient.disconnect();
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            log.info(String.format(UPLOAD_DIR_FAILURE, sourceDirectoryPath, targetDirectoryPath));
+        }
+        return isSuccess;
+    }
+
+    /**
+     * Recursively upload a directory to FTP server with the provided FTP client object.
+     *
+     * @param sourceDirectoryPath
+     * @param targetDirectoryPath
+     * @param logPrefix
+     * @throws IOException
+     */
+    private void uploadDirectory(final String sourceDirectoryPath, final String targetDirectoryPath,
+                                 final String logPrefix) throws IOException {
+        log.info(String.format(UPLOAD_DIR, logPrefix, sourceDirectoryPath, targetDirectoryPath));
+        final File sourceDirectory = new File(sourceDirectoryPath);
+        final File[] files = sourceDirectory.listFiles();
+        if (files == null || files.length == 0) {
+            log.info(logPrefix + "Empty directory at " + sourceDirectoryPath);
+            return;
+        }
+
+        // Make sure target directory exists
+        final boolean isTargetDirectoryExist = ftpClient.changeWorkingDirectory(targetDirectoryPath);
+        if (!isTargetDirectoryExist) {
+            ftpClient.makeDirectory(targetDirectoryPath);
+        }
+
+        final String nextLevelPrefix = logPrefix + "..";
+        for (File file : files) {
+            if (file.isFile()) {
+                uploadFile(file.getAbsolutePath(), targetDirectoryPath, nextLevelPrefix);
+            } else {
+                uploadDirectory(sourceDirectoryPath + "/" + file.getName(),
```

Comment:
Still use `/` for  `targetPath` because it is the path in remote FTP server. Using system `fileSeparator` will mess it up on Windows.